server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -333,7 +333,7 @@ public class IndicesService extends AbstractLifecycleComponent
     private final ScriptService scriptService;
     private final ClusterService clusterService;
     private final Client client;
-    private volatile Map<String, IndexService> indices = emptyMap();
+    private volatile Map<String, IndexService> indices = new ConcurrentHashMap<>();
```

Comment:
This being a `volatile ConcurrentHashMap` seems suspicious. Those two shouldn't normally go together.

Reply:
Good point. Have removed volatile from here.

Reply:
Good point. Removed volatile from here.

Reply:
Reverted this change.